core/src/main/scala/cats/Show.scala

```diff
@@ -27,7 +32,7 @@ object Show {
 
   final case class Shown(override val toString: String) extends AnyVal
   object Shown {
-    implicit def mat[A](x: A)(implicit z: Show[A]): Shown = Shown(z show x)
+    implicit def mat[A](x: A)(implicit z: ContravariantShow[A]): Shown = Shown(z show x)
```

Comment:
when I did a simple test in the repl this worked when I did `implicitly[Foo[_ >: A]]` did you try that form (not putting a specific type)? you never know with scala which minor translation of the code will work.